_posts/2020-02-14-creating-doc-assemblies.md

```diff
@@ -0,0 +1,181 @@
+I worked on the assembly locally, and then pushed the content to the Strimzi repo as a merge request.
+
+A new assembly requires:
+
+1. [An assembly file](https://github.com/strimzi/strimzi-kafka-operator/blob/master/documentation/assemblies/oauth/assembly-oauth.adoc)
+2. [Modules that contain the content](https://github.com/strimzi/strimzi-kafka-operator/tree/master/documentation/modules/oauth)
+
+So, the first thing I did was create the assembly (`assembly-oauth.adoc`) that would house the concepts and tasks related to OAuth 2.0.
+The assembly [describes how to use OAuth 2.0 token-based authentication](https://github.com/strimzi/strimzi-kafka-operator/blob/master/documentation/assemblies/oauth/assembly-oauth.adoc).
+This sits in its own oauth category folder (`oauth`) under the assemblies folder.
+Categories make it easier to find related material.
+
+````
+documentation
+└── assemblies
+      └── oauth
+            └── assembly-oauth.adoc
+````
+
+Assemblies are kept separate from the modules that form the content.
+Modules can include _concepts_ to provide understanding, _procedures_ to describe the steps to accomplish certain tasks, and related _reference_ data or information, often in lists or tables.
+
+As with each file that’s added to the documentation,
+the assembly file required a title and an ID that’s used whenever a cross-reference to the file is needed.
+We also add a comment to list the files that include an assembly or module:
+
+````
+// This assembly is included in the following assemblies:
+//assembly-deployment-configuration.adoc
+
+[id='assembly-oauth_{context}']
+= Using {oauth} token-based authentication
+````
+
+The _{context}_ suffix in the ID is a variable that is set at the `master.adoc` or assembly level,
+and is used to avoid duplication errors at build time when an assembly or module is reused.
+
+I could now build up the modules for the assembly. Again, I created a category for the modules, but this time under modules.
+
+````
+documentation
+└── modules
+      └──oauth
+           ├── con-oauth-authentication-broker.adoc
+           ├── con-oauth-authentication-client.adoc
+           ├── con-oauth-authentication-client-options.adoc
+           ├── con-oauth-authentication-flow.adoc
+           ├── con-oauth-config.adoc
+           ├── con-oauth-server-examples.adoc
+           ├── proc-oauth-broker-config.adoc
+           ├── proc-oauth-client-config.adoc
+           ├── proc-oauth-kafka-config.adoc
+           └── proc-oauth-server-config.adoc
+ ````
+
+The concepts files are prefixed with *con* and the procedures with *proc* to distinguish them. (A reference file gets a *ref*.) The concepts files provide context and an overview of what Oauth 2.0 is and how it operates. The split between the files is where the content naturally moves on to a new topic. Procedures describe a specific task related to Oauth 2.0. It might help to think of each of these files as representing a title in the table of contents.
```

Comment:
```suggestion
The concept files are prefixed with *con* and the procedures with *proc* to distinguish them. (A reference file gets a *ref*.) The concept files provide context and an overview of what Oauth 2.0 is and how it operates. The split between the files is where the content naturally moves on to a new topic. Procedures describe a specific task related to Oauth 2.0. It might help to think of each of these files as representing a title in the table of contents.
```